### ISSUE TITLE
Set `accessibilityLabel` attributes on form controls

### ISSUE THREAD

**Issue by ndarilek** [gcanti/tcomb-form-native#137]
### Version

Tell us which versions you are using:

- tcomb-form-native v0.4.0
- react-native v0.22.2

### Expected behaviour

Tell us what should happen

As a blind developer, I should be able to hear my screen reader read the text labels associated with fields.

### Actual behaviour

Tell us what happens instead

Field labels aren't read, as seems to be indicated by the README should happen.

### Steps to reproduce

1. Launch TalkBack (or presumably VoiceOver.)
2. Use a form.
3. Touch a text field.

This can likely be solved by setting the `accessibleLabel` attribute on form controls to the text of their associated labels. Should be easy to do this programatically since you're generating the label already.

https://facebook.github.io/react-native/docs/accessibility.html

For bonus points, set up a live region for form validation errors on Android so they are read out automatically. :)

Thanks, love working with this library.

**Comment by gcanti** [comment]
@ndarilek Thanks for pointing out. I'm not an expert on this matter, I'll be back to you as soon as I read the documentation and I have a plan (thanks for the link).

**Comment by alvaromb** [comment]
@gcanti we can use `Text`'s `accessible` prop or even the `accessibilityLabel` http://facebook.github.io/react-native/docs/accessibility.html#accessibility

If we find any problem, I know folks from FB's London office that work on accessibility and I can ping them.

**Comment by gcanti** [comment]
Usually for each single input we have 3 types of texts:

- label
- help
- error

The general rendering tree is:

```
<View>
  <Text>label</Text>
  <Input />
  <Text>help label</Text>
  <Text>validation error label</Text>
</View>
```

Now based on https://facebook.github.io/react-native/docs/view.html if we set the `accessible=true` attribute to the container `View` component we should obtain:

> By default, the label is constructed by traversing all the children and accumulating all the Text nodes separated by space.

So in our case the screen reader would read the concatenation of: label + help + error. I am right?

An example here https://code.facebook.com/posts/435862739941212/making-react-native-apps-accessible/ (the first Ads Manager app example)

In addition we could add the `accessibilityLiveRegion="polite"` (or `assertive` ??) attribute to the error `Text` component.

@ndarilek @alvaromb what do you think?

**Comment by ndarilek** [comment]
Sounds like a great idea. Thanks! I'll be happy to test out any branches 
once they are available.

I'm vaguely worried that this might convey a bit too much information 
for each field since what I was really after was only associating a 
label. What I'd mainly like is to know what field I'm typing into. But 
I'm willing to give this approach a shot.

**Comment by gcanti** [comment]
> I'll be happy to test out any branches once they are available.

This would be great, thanks.

**Plan B**

Another option would be just:

- set the `accessible=true` attribute of the container `View` component
- set the `accessibilityLabel` attribute of the container `View` component using the computed label as a value 
- add the `accessibilityLiveRegion="polite"` attribute to the error Text component

Not sure if `accessibilityLiveRegion` should be `assertive` instaed of `polite`.

I'm going to push 2 new branches:

- `a11y-a` with the first option (Plan A)
- `a11y-b` with the second option (Plan B)

I'll let you know when they are ready.

**Comment by alvaromb** [comment]
> Not sure if accessibilityLiveRegion should be `assertive` instead of `polite`.

I think it would be better to be `polite`. I don't know how `assertive` would handle multiple errors.

**Comment by gcanti** [comment]
@alvaromb ok, let's try with `polite`

**Comment by gcanti** [comment]
@ndarilek the new branches (`a11y-a`, `a11y-b`) are up

**Comment by ndarilek** [comment]
Wow, thanks for the quick turnaround! Here are some observations:

  * a:
    * Field labels were not spoken when focus landed on them.
    * Setting `accessible={true}` on the containing views was a bit 
confusing since that spoke the label whenever I touched anything within 
the bounds of the field's rectangle but *not* when I touched the text 
area. So basically if I touch outside of the text area (I.e. presumably 
in the screen rectangle the field is rendering in), I get the label, but 
not on the area itself.
  * Branch b appears to also behave the same.

Here's what I did to make a text field accessible, based on branch b 
because it's what I have checked out right now:

1. Removed `accessible={true}` from the form view.
2. Set `accessibilityLabel={locals.label}` on the `TextInput` element.

Now I get spoken feedback when I land on text fields. I assume the same 
will be true with other field types .

Unfortunately I don't have any error text yet to check out, but I'm a 
bit more worried about the labeled form fields and will be adding errors 
soon enough. :) When that happens I can provide feedback on the live 
regions.

Thanks so much for your work on this.

**Comment by gcanti** [comment]
@ndarilek In order to quickly test out an error message, just add an `error` option. Example:

```js
const Schema = t.struct({
  name: t.String
});

const options = {
  fields: {
    name: {
      error: 'Name is required'
    }
  }
};

...

<Form type={Schema} options={options} />
```

In the meanwhile I've put up a `a11y-c` branch (restarting from master) adding only `accessibilityLabel={locals.label}` to all inputs.

**Comment by ndarilek** [comment]
C works beautifully, and on my username/password form, clicking Login 
without filling it in nicely speaks both error messages in sequence.

Thanks!

**Comment by gcanti** [comment]
Great! 

@alvaromb when you find the time could you please double check my changes?

**Comment by alvaromb** [comment]
Sure! I'll test it tomorrow now that I'm exiting from the enormous amount of work we have. 

Enviado desde mi iPhone

**Comment by gcanti** [comment]
@ndarilek just released in https://github.com/gcanti/tcomb-form-native/releases/tag/v0.4.1
